nodeJS/express/deployment.md

```diff
@@ -0,0 +1,308 @@
+
+#### Domain names
+
+PaaS providers will give you a random domain name when you first deploy. In Heroku's case, it's usually something zen-like "afternoon-falls-4209". If you want to visit the app, you can go directly to `http://afternoon-falls-4209.herokuapp.com` to see your app live on the web in all its glory.
+
+The domain name will always be yours on a PaaS provider. They give each app a unique domain name that's yours as long as your app lives on their platform.
+
+In the real world, you'll want to link it to your own custom domain, for example, `http://mycooldomain.com`. It's worth mentioning that you don't need custom domain names for the portfolio projects you'll be building in this course. The random domains your PaaS provider gives you will be good enough. However, if you have an itch for tricking out your app with a custom domain name, you'll first need to purchase a domain from a registrar like [Porkbun](https://porkbun.com/) or [NameSilo](https://www.namesilo.com/).
+
+To find a new domain, try using [Domainr](https://domainr.com/).
+
+Once you have your domain, you need to point it to your project. The provider you are using will have exhaustive documentation on using custom domain names on their platform.
+
+### Our recommended PaaS services
+
+Choosing a PaaS provider was once a simple decision. Heroku had a free tier that gave you everything needed to host as many small apps as you wanted, but they unfortunately discontinued it in 2022.
+
+Luckily, there are still plenty of other great options out there. The downside is that they all have very limited free tiers. For this reason, and to accommodate as many of our learners as possible, we're going to recommend a range of options instead of just one.
+
+You can host most of your projects for free using a combination of the different providers we recommend, but you'll have a bit more leg work to do. You'll have to create an account and get familiar with a few different providers.
+
+If you can pay for hosting, things are more straightforward. It will afford you the luxury of choosing one provider you can learn deeply, and you'll be able to manage all your apps from one place.
+
+Whatever your circumstances, we've got you covered. Here are the PaaS providers we recommend.
+
+---
+
+#### Fly.io
+
+- Fly.io uses a convenient CLI tool for deployment.
+- Pay for what you use with very reasonable rates. Each app should cost around $4 per month.
+- $20 a month should be enough to host eight apps (including three apps for free).
+- Fly.io charges $0.15/GB of RootFS for machines stopped for 30 days.
+
+##### Fly.io: Free Plan
+
+- You can host three apps for free before you need to start paying.
+- Fly.io waives monthly invoices that total less than $5 USD. So, although it may look like
+  you are being charged, it's very unlikely that you will exceed $5 and actually have to pay.
```

Comment:
Could the verbiage for this part be changed to match the updated text in #28036 (inc. pending changes from comments)? Just the hobby plan bit, not the legacy hobby plan bit.

```suggestion
##### Fly.io: Hobby Plan

- New customers get a one-time $5 free trial credit to test Fly.io at no cost. The credit will only start being used if you exceed the free resource allowance, and once it's been used up, you will be automatically placed on the $5/month Hobby plan.
- The longevity of your free trial credit depends on how many resources you consume. More complex apps with more traffic may consume all free resources within a month, whereas simpler apps may last longer.
```

Reply:
**Update:** #28199 made further changes to the Fly.io section, due to *even more* recent changes to their free trial. The current lesson content for Fly.io is the latest, so if you could use the text from the current lesson, that'd be great!